### DISCUSSION TITLE
[Rancher Desktop] nerdctl compose up = no such network

### DISCUSSION THREAD

**Discussion by johnmanko** [containerd/nerdctl#3250] [Q&A]
_Asked over at [rancher-sandbox/rancher-desktop](https://github.com/rancher-sandbox/rancher-desktop/discussions/7250), but haven't received an answer._

I'm working on a k8s cluster using Rancher Desktop, but want to set up my databases external the cluster.  I've used this same compose.yaml with Docker Desktop, but with Rancher Desktop I wanted to eliminate docker and just work with nerdctl and containerd (so I can install cilium CNI and choose the version of kubernetes). 

When doing a `nerdctl compose up`, I get the following error:

```
INFO[0000] Creating network postgres-stack_db-net
INFO[0000] Creating network postgres-stack_default
INFO[0000] Creating volume postgres-stack_pgdata
INFO[0000] Ensuring image postgres:16.3-alpine3.20
INFO[0000] Ensuring image adminer:4.8.1-standalone
INFO[0000] Creating container postgres-stack-db-1
FATA[0000] failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: time="2024-07-24T21:09:39-04:00" level=fatal msg="no such network: \"postgres-stack_db-net\""
Failed to write to log, write /var/lib/nerdctl/dbb19c5e/containers/default/2c005e1d9f4c16783a57c7b2c8822a800ec46230be733cb62410697c6fc2c7eb/oci-hook.createRuntime.log: file already closed: unknown
FATA[0000] error while creating container postgres-stack-db-1: exit status 1
Error: exit status 1
```

The `compose.yaml` is 

```yaml
name: postgres-stack

services:

  db:
    image: postgres:16.3-alpine3.20
    restart: always
    ports:
      - 5432:5432
    environment:
      POSTGRES_PASSWORD: postgres
      PGDATA: "/var/lib/postgresql/data/pgdata"
    volumes:
      - pgdata:/var/lib/postgresql/data
      - "./docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d"
    networks:
      - db-net

  adminer:
    image: adminer:4.8.1-standalone
    restart: always
    ports:
      - 2345:8080
    networks:
      - db-net

volumes:
  pgdata: {}

networks:
  db-net: {}
```

My RD setup is configured to use containerd, not docker.  I also disabled the default CNI and Ingress controller and installed cilium, not that it should matter.

When trying to create the missing network with nerdctl, it appears to complete, but nothing is created.

```shell
❯ nerdctl network create postgres-stack_db-net
f26c657d0c1a7c5f30e66f760cab93c9fb20a723e0c20262cc3816a5990d34fc

❯ nerdctl network ls
NETWORK ID    NAME      FILE
              cilium    /etc/cni/net.d/05-cilium.conflist
              host
              none
```

```shell
❯ which nerdctl
/Users/john/.rd/bin/nerdctl
```

What am I missing?

**Comment by AkihiroSuda** [comment]
> When trying to create the missing network with nerdctl, it appears to complete, but nothing is created.

This looks weird, and probably specific to Rancher Desktop?
Then this issue should be reported to Rancher Desktop, not to nerdctl.

**Comment by RonBarkan** [comment]
I am getting the same behavior with generic Kubernetes + Cilium install.

Using `nerdctl create network` followed by `network ls` or `network inspect` shows that there's no network, even through the create completed successfully.

I suspect Cilium took over.
